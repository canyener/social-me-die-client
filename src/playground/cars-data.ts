export interface ICar {
  color: string,
  model: string,
  topSpeed?:number
}

const car1: ICar = {
  color: 'blue',
  model: 'BMW',
}

const car2: ICar = {
  color: 'red',
  model: 'Mercedes',
  topSpeed: 100
}

export const cars = [car1, car2]