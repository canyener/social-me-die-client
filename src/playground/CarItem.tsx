import React, { FC } from 'react'

import { ICar } from './cars-data'

interface IProps {
  car: ICar
}

const CarItem: FC<IProps> = ({ car }) => {
  return (
    <div>
      <h1>{car.color}</h1>
    </div>
  )
}

export default CarItem