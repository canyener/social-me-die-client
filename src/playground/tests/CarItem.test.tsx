import React from 'react' 
import { shallow } from 'enzyme'

import { ICar } from '../cars-data'
import CarItem from '../CarItem'

test('Should render CarsItem correctly', () => {
  const car: ICar = {
    color: 'test color',
    model: 'test model',
    topSpeed: 333
  }
  const wrapper = shallow(<CarItem car={car} />)

  expect(wrapper).toMatchSnapshot()
})