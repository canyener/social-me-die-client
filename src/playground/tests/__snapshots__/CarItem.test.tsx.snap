// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CarsItem correctly 1`] = `
<div>
  <h1>
    test color
  </h1>
</div>
`;
